### ISSUE TITLE
Add rtd1296 support to some packages (Sonarr/ffmpeg/SABnzbd/etc) for DSx18/D218play/DS418j

### ISSUE THREAD

**Issue by jfmou** [SynoCommunity/spksrc#3079]
_For new Package Requests, see the guidelines_  

### Setup
_Package Name:_ Sonarr/ffmpeg/Radarr/SABnzbd/Python V2/nano/Jackett/Git
_Package Version:_

_NAS Model:_ DS118, DS218, DS218play, DS418j, DS418
_NAS Architecture:_ rtd1296
_DSM version:_ DSM 6.1.4-15217 Update 5

### Expected behavior
I'd like to setup sonarr on my ds218 and transcode DTS via ffmpeg.

### Actual behavior

Available Packages :  

1. CouchPotato
2. CouchPotato Custom
3. Full-Text RSS
4. Headphones
5. Headphones Custom
6. HTPC-Monitor
7. Jappix
8. LazyLibrarian
9. Maraschino
10. NZDmegasearch
11. Salt Master
12. SaltPad
13. SickBeard Custom
14. Subliminal

Thank you for your time and sorry if this is a duplicate of an already existing issue. (I've used [#2945](https://github.com/SynoCommunity/spksrc/issues/2945) to start the listing of missing packages.)

**Comment by WodKa78** [comment]
[here](https://github.com/WodKa78/-syno-packages) you can find what i was able to compile for own use. All this comes with no support, i am not a programmer of any kind.

**Comment by Mike981874** [comment]
I would love to see Sonarr available for DS218+ also. Bought one and didn't realize it wasn't updated for this version/model of the device yet.

**Comment by WodKa78** [comment]
unrelated to this issue, check [here](https://github.com/SynoCommunity/spksrc/issues/2945#issuecomment-357194829) as are tons of work being done for your nas

**Comment by Safihre** [comment]
New package list here (just added Sonarr/Radarr): https://github.com/SynoCommunity/spksrc/issues/2945#issuecomment-357008518

**Comment by jfmou** [comment]
thanks @Safihre ! I'll try them and give you feedback asap.

**Comment by jfmou** [comment]
Ok so good so far, i've successfully installed python, git, mono and sonarr 🔥 
My workplace actually block custom port so I can't try sonarr right now, i'll update you right back home.

**Comment by ymartin59** [comment]
@m4tt075 Concerning `ffmpeg` I really think about excluding `libx265` (and even `x264`) from `rtd1296` package as far as assembly code is ported upstream... but it sounds like irrelevant concerning usability then.

**Comment by jfmou** [comment]
it's been 2 days now that I use jackett, sonarr, mono, git et python on my ds218, and it works like a charm :)

**Comment by m4tt075** [comment]
@ymartin59 I have just figured out that you can make `x264` compile with the following (platform-specific) `configure` arguments:
```
ifeq ($(findstring $(ARCH), $(ARM8_ARCHES)),$(ARCH))
ENV += x264_ARCH=aarch64
CONFIGURE_ARGS += --disable-asm --disable-cli
endif
```
So if we can live without cli, we might be fine. Pretty sure `tvheadend` does not need it, as the cli is disabled in the statically linked `x264` version of `tvheadend` as well, but not sure about breakage of other packages. What do you think?

**Comment by ymartin59** [comment]
As owner of DSM rtd1296 architecture, may you please download and test this "beta" `ffmpeg` built: http://dl.free.fr/oIHqgBdrA
If possible, please run from command line `ffmpeg -i` on various media files. 
Warning: do not expect Video Station to use it, as recent version is stuck to Synology embedded version.
Please report success or any failure with command output

**Comment by silv3r23** [comment]
Hello, when can we expect the packages for: radarr, sonarr, sabnzbd to be available in the package center of DS418j (DSM 6.1.4-15217 Update 5)?

**Comment by Safihre** [comment]
@silv3r23 Please use the test-packages listed here for now! https://github.com/SynoCommunity/spksrc/issues/2945#issuecomment-357008518

You will have to manually install them for now.

**Comment by jfmou** [comment]
I tried to install LazyLibrarian available on package center on my DS218, and got an error saying python is missing from package center even if I installed it manually from [here](https://github.com/Safihre/spksrc/releases/tag/python-dsm61-new-all).

**Comment by jfmou** [comment]
After updating to DSM DSM 6.1.5-15254, I've got the folowing error in jackett : 

>  The mono certificate store is not initialized.
Please run the following command as root:
cert-sync /etc/ssl/certs/ca-certificates.crt
If you don't have root access, please run the following command as the jackett user (sc-jackett):
cert-sync --user /etc/ssl/certs/ca-certificates.crt

Every tests of indexers failed.
I've been able to make it work again by simple doing : 
`sudo /var/packages/mono/target/bin/cert-sync /etc/ssl/certs/ca-certificates.crt 
`

This way I still got the error message, but indexers tests now work. I've no idea if there is any correlation between dsm 6.1.5 and this problem, though. I just thought I should share the facts.

**Comment by m4tt075** [comment]
@jfmou Could you please post which versions of jackett and mono you were running before you touched anything (or if you started from scratch) and then, to which versions of jackett and mono you have been updating in which sequence? Thanks.

**Comment by m4tt075** [comment]
@jfmou Or has it only been the upgrade of your DSM which triggered the error?

**Comment by ymartin59** [comment]
@m4tt075 I disagree. Mono installation only requires one single execution of `cert-sync` to copy system certificates into runtime store. From my point of view, either command has not been triggered at package installation (maybe because it was an upgrade) or command has failed (silently ?)... Is there any clue in `install.log` ?

**Comment by m4tt075** [comment]
@ymartin59 In the 5.4.x as well as the 5.8.0 package installer script `/var/packages/mono/target/bin/cert-sync /etc/ssl/certs/ca-certificates.crt > /dev/null` should be executed during `postinstall` as well as `postupgrade`, but let's see...

**Comment by ymartin59** [comment]
Is there an easy way to test that a Mono installation already has proper certificates in its store, to check `cert-sync` has run.

**Comment by m4tt075** [comment]
Good question. I don't know. But I have just figured out that rerunning `/var/packages/mono/target/bin/cert-sync /etc/ssl/certs/ca-certificates.crt` doesn't harm. The command does not add certificates it already trusts, but does add new ones incrementally.

**Comment by jfmou** [comment]
@m4tt075 I run mono 5.4.1.6-9 and jackett 0.8.490-5 since safhire made them available and everything worked flawlessly. I didn't touch anything on their parts.
2 days ago, I updated to DSM 6.1.5 at 5:14pm.
Sonarr started to output this at 5:16pm :
```json 
18-2-2 17:16:20.3|Error|DSMInfoProxy|Failed to get serial number. Reason: 119 - Unknown error
18-2-2 17:16:20.3|Warn|SerialNumberProvider|Could not get the serial number from Download Station 127.0.0.1:5000
[v2.0.0.5085] NzbDrone.Core.Download.Clients.DownloadClientException: Failed to get serial number. Reason: 119 - Unknown error
  at ...
```

 and the first trace of ssl errors I got happened at 5:22pm

```json
18-2-2 17:22:53.9|Warn|HttpClient|HTTP Error - Res: [GET] http://192.168.1.200:9117/api/v2.0/indexers/thepiratebay/results/torznab/api?t=tvsearch&cat=5030,5040&extended=1&apikey=(removed)&offset=0&limit=100: 500.InternalServerError  
{
  "result": "error",
  "error": "Exception (thepiratebay): Ssl error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED\n  at /spksrc/spk/mono/work-rtd1296-6.1/mono-5.4.1.6/external/boringssl/ssl/handshake_client.c:1132: An error occurred while sending the request",
"stacktrace": "..."  
}
```

Can't find anything in logs from jackett
Once I added every certificates manually and got a success message like `172 certificates successfully imported` or something close to this, every sonarr / jackett errors disappeared.

**Comment by m4tt075** [comment]
Thanks for the detailed report, @jfmou .

@ymartin59 This is not related to the `Mono 5.8.0` corlib issue. Only idea I have is that the DSM update changed the `ca-certificates.crt` file and that this requires a re-synchronisation with mono. Does that make sense to you? Or do you have another idea?

**Comment by ymartin59** [comment]
@jfmou May you please detail what you did to add "every certificates manually" ?

@m4tt075 Well, C# and Mono are not my cup of tea... even if I am comfortable with certificate management, there is something weird here.

**Comment by jfmou** [comment]
@ymartin59, I've run the following over ssh on my ds218 :
```bash
sudo /var/packages/mono/target/bin/cert-sync /etc/ssl/certs/ca-certificates.crt
```

**Comment by Safihre** [comment]
The new `mono` package was released that now preserves the certificates after updates.
Other packages for `rtd1296` are slowly added, test packages are available in #3138.
Specific requests for packages can be made in their respective issues.
Many of the packages listed in this post are now either available officially or in #3138 for testing. After the official release on SynoCommunity, the package will auto-update from the official release.
Therefore closing this.
